crates/sui-network/src/discovery/builder.rs

```diff
@@ -158,8 +164,8 @@ impl UnstartedDiscovery {
         )
     }
 
-    pub fn start(self, network: anemo::Network) -> Handle {
-        let (event_loop, handle) = self.build(network);
+    pub fn start(self, network: anemo::Network, keypair: NetworkKeyPair) -> Handle {
+        let (event_loop, handle) = self.build(network, keypair);
```

Comment:
can we make sure that this is the same keypair as is the peerid in the network by comparing the bytes of the peerid and the pubkey?

Reply:
good idea, done